README
SimpleMash v1.0.17 RC3
J.Rocela (me@iamjamoy.com)

Thank you for Downloading SimpleMash, now you can get Feeds
from your favorite sites and post them to your Blogs. Gets answers
from Yahoo! Answers

== Installation ==
1. Unzip files to your Wordpress Plugin folder
2. Install the plugin from your Wordpress Administration Panel
3. Access the Admin panel from Tools > SimpleMash
4. Start by adding a feed from the ADD tab.
5. Repeat step 4 until Satisfied :)

For more information, please email me@iamjamoy.com. I am still
working on a documentation page, or it's own project page on 
gurucs.com or iamjamoy.com.

For ISSUES, CONCERNS, BUG REPORTS, FEATURE REQUESTS and SUGGESTIONS
please contact me@iamjamoy.com or jamoy@gurucs.com

CHANGELOG
SimpleMash v1.0.17 RC3
J.Rocela (me@iamjamoy.com)

1.0.5 Initial Release
1.0.6 Added support for More RSS Feeds.
      Altered database schema
	  Added PHP4 Support
	  Changed Installation Procedure
1.0.6 RC1 More stable than the last release
	  Added JSON Support
	  Altered Client Code
1.0.7 RC2 

1.0.8 RC2 

1.0.9 RC2 

1.0.12 RC3
	Fixed Problems on Publishing
	Category Fixes
	Aggregation Method Fixes
	Settings are moved from the global scope to individual scope
	Added the Clear Entries Functionalities
	Initial SimpleMash Y! implementation
1.0.13 RC3
	Added Post Deletion hooks
1.0.15 RC3
	Implemented Yahoo! Answers integration.
	Removed Double Aggregation
	Fixed option checkboxes.
1.0.16 RC3
	Added condition to prevent double publication
1.0.17 RC3
	Testing the New Simplemash Logic

simplemash.php
<?php
/**
 * Plugin Name: 	SimpleMash RSS Aggregator
 * Plugin URI:	http://www.gurucs.com/products/simeplemash
 * Description:	Import Feeds from any URL you like and narrow them down with a simple Keyword Filter and attach your Feeds to categories. Manually Approve Feeds or automatically dispaly them to your blog.
 * @version:		1.0.17 RC3
 * Author: 		J. Rocela (me@iamjamoy.com)
 * Author URI:	http://www.gurucs.com
 *
 * copyright (c) 2009  J. Rocela(me@iamjamoy.com)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 *
 */

define ('SIMPLEMASH_VERSION','1.0.17 RC3'); //define version number

/**
 * simplemash_install
 *
 * installs the plugin's database tables and configurations
 *
 * @return void
 */
function simplemash_install(){
	global $wpdb;
		
	$site_table = $wpdb->prefix . "simplemash_sites";
	$sql = "CREATE TABLE IF NOT EXISTS `$site_table` (
		`id` int(11) NOT NULL auto_increment,
		`feed_url` text NOT NULL,
		`site_url` text NOT NULL,
		`site_name` char(128) NOT NULL,
		`description` char(255) NOT NULL,
		`last_update` char(10) NOT NULL,
		`next_update` char(10) NOT NULL,
		`updated_times` bigint(10) NOT NULL default '1',
		`update_count` bigint(10) default NULL,
		`filter_from_title` char(255) default NULL,
		`filter_from_body` char(255) default NULL,
		`filter_from_description` char(255) default NULL,
		`filter_from_category` char(255) default NULL,
		`category_id` bigint(20) default NULL,
		`publish_automatically` tinyint(1) NOT NULL default '1',
		`show_source` tinyint(1) NOT NULL default '1',
	    `gather` int(20) NOT NULL default '0',
	    `frequency` int(20) NOT NULL default '60',
	    `keep` int(20) NOT NULL default '5',
	    `post_id` bigint(20) default NULL,
		PRIMARY KEY  (`id`)
	) ENGINE=InnoDB  DEFAULT CHARSET=utf8 ;";
	$wpdb->query($sql);
	$entries_table = $wpdb->prefix . "simplemash_entries";	
	$sql = "CREATE TABLE IF NOT EXISTS `$entries_table` (
		`id` int(11) NOT NULL auto_increment,
		`site_id` int(11) NOT NULL,
		`hash` char(64) NOT NULL,
		`entry_url` text NOT NULL,
		`title` char(255) NOT NULL,
		`description` text,
		`body` text,
		`datetime` char(64) NOT NULL,
		`acquired` char(10) NOT NULL,
		`approved` tinyint(1) NOT NULL,
		`published` bigint(20) NOT NULL default '0',
		`time_published` char(10) default NULL,
		`category_id` bigint(20) default NULL,
		PRIMARY KEY  (`id`)
	) ENGINE=InnoDB  DEFAULT CHARSET=utf8 ;";
	$wpdb->query($sql);	
}

/**
 * @ignore
 */
function simplemash_uninstall(){	
	// PRESERVE DATA
	/**
	global $wpdb;
	$site_table = $wpdb->prefix . "simplemash_sites";
	$sql = "DROP TABLE $site_table;";
	$wpdb->query($sql);
	$entries_table = $wpdb->prefix . "simplemash_entries";	
	$sql = "DROP TABLE $entries_table;";
	$wpdb->query($sql);
	*/
}


register_activation_hook(__FILE__,'simplemash_install');
register_deactivation_hook(__FILE__,'simplemash_uninstall');

/**
 * require the simplemash class
 */
require_once "simplemash.class.php";

/**
 * initialize the simplemash class
 *
 * this method will check the plugin's requirements and the server's environment.
 */
$sm = new SimpleMash();

/**
 * use the simplemash fangs, i mean hooks
 *
 * this method will call the hooks associated to the simeplmash class. this is the most crucial method
 * for wordpress to use this plugin. *do not delete*
 */
$sm->wp_hook();

?>

simplemash.ajax.php
<?php
/**
 * AJAX Actions Bridges to the Simplemash API
 *
 * @author: 	John Rocela(me@iamjamoy.com)
 * @author_uri:	http://iamjamoy.com
 * @copyright:	Guru Consultation Services http://gurucs.com
 * @version:		1.0.17 RC3
 * @package: 	SimpleMash
 *
 * copyright (c) 2009  J. Rocela(me@iamjamoy.com)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 *
 */

// Get the Action String
$request = @$_POST['action'];
// Require the SimpleMash API
require_once 'simplemash.class.php';

// Instatiate the SimpleMash API
$fg = new SimpleMash();

// Switch case for the API Actions
switch($request){
	case 'checkfeed':
		$fg->simplemash_ajax_checkfeed($_POST['url']);
	break;
	case 'previewfeed':
		$fg->simplemash_ajax_previewfeed($_POST['url'],$_POST['filter']);
	break;
	case 'addfeed':
		$fg->simplemash_ajax_addfeed($_POST['url'],$_POST['title'],$_POST['filter'],$_POST['category'],$_POST['automatic'],$_POST['source'],$_POST['gather'],$_POST['frequency'],$_POST['keep'],@$_POST['addCategory'],@$_POST['addCategoryXML']);
	break;
	case 'fetchItem':
		$fg->simplemash_ajax_fetchItem($_POST['id']);
	break;
	case 'saveItem':
		$fg->simplemash_ajax_saveItem($_POST['id']);
	break;
	case 'deleteItem':
		echo $fg->simplemash_ajax_deleteItem($_POST['id']);
	break;
	case 'cleanEntries':
		echo $fg->simplemash_ajax_cleanentries();
	break;
	case 'fetchEntries':
		echo $fg->simplemash_ajax_fetchEntries($_POST['condition'],@$_POST['true']);
	break;
	case 'fetchEntry':
		echo $fg->simplemash_ajax_fetchEntry($_POST['id']);
	break;
	case 'saveEntry':
		echo $fg->simplemash_ajax_saveEntry($_POST['id'],$_POST['info']);
	break;
	case 'aggregateFeeds':
		echo $fg->simplemash_ajax_aggregateFeeds($_POST['feeds']);
	break;
	case 'deleteFeeds':
		$fg->simplemash_ajax_deleteFeeds($_POST['feeds']);
	break;
	case 'saveInfo':
		echo $fg->simplemash_ajax_saveInfo($_POST['info']);
	break;
	case 'deleteEntry':
		echo $fg->simplemash_ajax_deleteEntry($_POST['id']);
	break;
	case 'deleteEntries':
		echo $fg->simplemash_ajax_deleteEntries($_POST['entries']);
	break;
	case 'approveEntry':
		echo $fg->simplemash_ajax_approveEntry($_POST['id']);
	break;
	case 'approveEntries':
		$fg->simplemash_ajax_approveEntries($_POST['entries']);
	break;
	case 'fetchFeeds':
		echo $fg->simplemash_ajax_fetchFeeds();
	break;
	case 'deleteEntry':
		$fg->simplemash_ajax_deleteEntry($_POST['id']);
	break;
	case 'clearEntries':
		echo $fg->simplemash_ajax_clearEntries($_POST['id']);
	break;
}

?>